### PR TITLE
🙌 - Added option for custom extra headers in Websocket Transport

### PR DESCRIPTION
**Pull request by Adrxx**

### CODE REVIEW

File: MQTTClient.podspec [hunk 1]
```diff
@@ -1,13 +1,13 @@
 Pod::Spec.new do |mqttc|
 	mqttc.name         = "MQTTClient"
-	mqttc.version      = "0.14.0"
+	mqttc.version      = "0.14.1"
```
Comment (MikeWeller) [inline MQTTClient.podspec]: I'm guessing this should be an increment of the minor version, i.e. bump to 0.15.0 (or maybe that would be another PR for a release) since this is a feature enhancement rather than a bug fix.

File: MQTTClient.podspec [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
 	mqttc.summary      = "iOS, macOS and tvOS native ObjectiveC MQTT Client Framework"
 	mqttc.homepage     = "https://github.com/novastone-media/MQTT-Client-Framework"
 	mqttc.license      = { :type => "EPLv1", :file => "LICENSE" }
 	mqttc.author       = { "novastonemedia" => "ios@novastonemedia.com" }
 	mqttc.source       = {
-		:git => "https://github.com/novastone-media/MQTT-Client-Framework.git",
-		:tag => "0.14.0",
+		:git => "https://github.com/ment-mx/MQTT-Client-Framework.git",
```
[the remark this answers is not in the document]
Comment (mendirattanishant) [reply MQTTClient.podspec]: I agree to Mike!

**Comment by Adrxx** [comment]
I have fixed all the problems now, what do you think? ready to merge?

**Comment by Adrxx** [comment]
By the way. For somme reason if I use the pod like this locally, it crashes with:
`-[MQTTWebsocketTransport setUrl:]: unrecognized selector sent to instance` I assume it has something to do with another commit #461. What happens in this case? should I go back to tag 0.14.0 and commit my changes there to another branch? Sorry, I'm really new to pull requests hahaha.

**Comment by jcavar** [comment]
@Adrxx can you just merge master to your branch? I think that should fix it.

File: MQTTClient/MQTTClient/MQTTWebsocketTransport/MQTTWebsocketTransport.h [hunk 3]
```diff
@@ -49,6 +49,12 @@
  * defaults to nil
  */
 @property (strong, nonatomic) NSArray *pinnedCertificates;
+  
+/** additionalHeaders an NSMutableDictionary containing extra headers sent when stablishing the websocket connection. Useful for custom authorization protocols. e.g. AWS IoT Custom Auth.
```
Comment (jcavar) [inline MQTTClient/MQTTClient/MQTTWebsocketTransport/MQTTWebsocketTransport.h]: This should be `NSDictionary`to reflect actual type.

**Comment by Adrxx** [comment]
Done! @jcavar Also now everything is merged to master, I think I have finally fixed everything 😄

File: MQTTClient/MQTTClient/MQTTWebsocketTransport/MQTTWebsocketTransport.h [hunk 4]
```diff
@@ -44,6 +44,12 @@
  * defaults to nil
  */
 @property (strong, nonatomic) NSArray *pinnedCertificates;
+  
+/** additionalHeaders an NSDictionary containing extra headers sent when stablishing the websocket connection. Useful for custom authorization protocols. e.g. AWS IoT Custom Auth.
```
Comment (jcavar) [inline MQTTClient/MQTTClient/MQTTWebsocketTransport/MQTTWebsocketTransport.h]: I think another typo here: "stablishing"

**Comment by jcavar** [comment]
Thank you @Adrxx!

**Comment by Adrxx** [comment]
You're welcome @jcavar! Sorry, I missed that one. Are you gonna make the correction of the typo? do you have final control of what gets merged? Let me know If I can help. By the way, I did this because the AWS IoT sampled lacked this functionality, then I realized their code https://github.com/awslabs/aws-sdk-ios-samples/tree/master/IoT-Sample/Swift is kind of a copy-paste of yours. Did you already know that?

**Comment by jcavar** [comment]
No, don't worry I will do it.

Which part is copy paste?

**Comment by Adrxx** [comment]
@jcavar Ok so turns out the copy-paste is actually from SocketRocket. Apparently AWS made its own `SRWebSocket` Class called `AWSSRWebSocket`.  You can see it's copy-pasted because the comments are all the same. Comments like:  
```objective-c
    NSAssert(self.readyState != SR_CONNECTING, @"Invalid State: Cannot call send: until connection is open");
    // TODO: maybe not copy this for performance
    data = [data copy];
```
I mean, I just tought it was interesting. I don't think there's anything wrong with it.
